public/index.html

```diff
@@ -144,6 +144,13 @@
 
 
 <div id="contentWrapper" class="container-fluid">
+
+
+    <div id="loadingWheel">
```

Comment:
Why not attach the event to this element instead of `document` or `window` ?  Not sure how that affects dispatching, it might not be worth it.